kernel/libs/aster_page_allocator/src/buddy_allocator.rs

```diff
@@ -0,0 +1,251 @@
+// SPDX-License-Identifier: MPL-2.0 OR MIT
+// To be specific, the original source code is from [buddy_system_allocator](https://github.com/rcore-os/buddy_system_allocator),
+// which licensed under the following license.
+//
+// SPDX-License-Identifier: MIT
+//
+// Copyright (c) 2019-2020 Jiajie Chen
+//
+// We make the following new changes:
+// * Add `alloc_specific` to allocate a specific range of frames.
+// * Implement `PageAlloc` trait for `BuddyFrameAllocator`.
+// * Refactor API to differentiate count and size of frames.
+//
+// These changes are released under the following license:
+//
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::collections::btree_set::BTreeSet;
+use core::{alloc::Layout, array, cmp::min};
+
+use ostd::mm::{page::allocator::PageAlloc, Paddr, PAGE_SIZE};
+
+/// Buddy Frame allocator is a frame allocator based on buddy system, which
+/// allocates memory in power-of-two sizes. The implementation of this allocator
+/// originated from crate [buddy_system_allocator](https://crates.io/crates/buddy_system_allocator).
```

Comment:
Remove the link to the source in the documentation as it is specified in the license preamble.